scripts/asm-installer/install_asm

```diff
@@ -1977,6 +1982,26 @@ EOF
   outro
 }
 
+establish_managed_logging_bucket() {
+  # The only supported regions are: global, us-central1, us-east1, us-west1, asia-east1, europe-west1.
+  BUCKET_LOCATION="global"
+  case "$CLUSTER_LOCATION" in
+    us-central*    ) BUCKET_LOCATION="us-central1";;
+    us-east*       ) BUCKET_LOCATION="us-east1";;
+    us-west*       ) BUCKET_LOCATION="us-west1";;
+    asia*          ) BUCKET_LOCATION="asia-east1";;
+    europe*        ) BUCKET_LOCATION="europe-west1";;
+    *              ) BUCKET_LOCATION="global";;
+  esac
```

Comment:
This case statement bugged me a bit and I was thinking of a way to avoid it, and then I ended up implementing it to prove to myself that it worked:

```
pref_len() {
  local LEN; LEN="${#1}";
  if ((${#2} > $LEN)); then
    LEN="${#2}"
  fi

  for ((i=0; i<$LEN; ++i)); do
    if [[ "${1:i:1}" != "${2:i:1}" ]]; then
      echo "${i}"
      return
    fi
  done

  echo "${LEN}"
}

find_best_region() {
  REGION="global"
  BEST=0
  for location in $(gcloud beta logging locations list --format='value(location_id)'); do
    SCORE="$(pref_len "${CLUSTER_LOCATION}" "${location}")"
    if (( $SCORE > $BEST )); then
      BEST="${SCORE}"
      REGION="${location}"
    fi
  done

  echo ${REGION}
}
```

Then we can just do `BUCKET_LOCATION="$(find_best_region)"` and drop the case statement, and it also avoids hard coding strings. (And hopefully this script doesn't live that long, but in the case that GCS adds another region we're future proof.)

Reply:
ok. as long as i don't have to maintain that bit of bash-fu :). updated.

Reply:
It looks less scary like this:

```python
def pref_len(a, b):
  len = len(b) len(a) > len(b) else len(a)
  for i in range(len):
    if a[i] != b[i]:
      return i

def find_best_region(target):
  region="global"
  best=0
  for location in locations:
    cur_val = pref_len(location, target)
    if cur_val > best:
      best=cur_val
      region=location

  return location
```

Reply:
case statement replaced.